### DISCUSSION TITLE
@entangle an array of Eloquent Models lose their type

### DISCUSSION THREAD

**Discussion by crazedVic** [alpinejs/alpine#1109] [4. General]
So first off, I understand why this is happening, just curious if anyone has figured out a neat workaround.

Basically I have this in my livewire blade view: 
`<div x-data="{selected: @entangle('selectedVehicles')}">`

And I have this in my livewire component: 
` public $selectedVehicles = [];`

I use a livewire event to add vehicles to this collection:
```
public function add(Vehicle $vehicle){
        $this->selectedVehicles[] = $vehicle;
    }
```
User clicks add on vehicle in a list of all vehicles, the vehicle gets added to the array via livewire action, then alpinejs automatically updates the selected vehicles list using x-for.  This works great.  There's a delete button on the selected vehicle list items where I use alpinejs to remove from the **selected** array.  It stays well in sync with the component array.

This all works swimmingly until I go to save the selectedVehicles to the user.
`Call to a member function save() on array`

By @entangling the array, the objects therein are stripped of their _type_ so to speak. 

So then I loop through the $this->selectedVehicles and cast them back to Vehicle _type_ and then they are Eloquent models once more and all is right in the world.

So does anyone know if there's a way to preserve the array item's _type_ whilst still using @entangle?

**Comment by CarstenRuetz** [comment]
same here, alpine/livewire can keep types of an Eloquent Collection, but with entangle the Eloquent Collection type is not kept. Any ideas?
